front/src/modules/trainschedule/helpers/pacedTrain.ts

```diff
@@ -1,9 +1,93 @@
-/* eslint-disable import/prefer-default-export */
-import type { PacedTrainWithDetails } from '../components/Timetable/types';
+import type { PacedTrainException } from 'common/api/osrdEditoastApi';
+import type { OccurrenceId } from 'reducers/osrdconf/types';
+import {
+  extractExceptionIdFromOccurrenceId,
+  extractOccurrenceIndexFromOccurrenceId,
+  isIndexedOccurrenceId,
+} from 'utils/trainId';
+
+import { exceptionChangeGroupsDict } from '../components/Timetable/consts';
+import type {
+  ExceptionChangeGroup,
+  OccurrenceException,
+  PacedTrainWithDetails,
+} from '../components/Timetable/types';
 
 export const getOccurrencesNb = ({ timeWindow, interval }: PacedTrainWithDetails['paced']) => {
   if (interval.ms === 0) {
     throw new Error('Interval cannot be 0');
   }
   return Math.ceil(timeWindow.ms / interval.ms);
 };
+
+/**
+ * Based on an exception list and an occurrence id, find the corresponding exception
+ */
+export const findExceptionWithOccurrenceId = (
+  exceptions: PacedTrainException[],
+  occurrenceId: OccurrenceId
+) => {
+  if (isIndexedOccurrenceId(occurrenceId)) {
+    const occurrenceToUpdateIndex = extractOccurrenceIndexFromOccurrenceId(occurrenceId);
+
+    return exceptions.find((exception) => exception.occurrence_index === occurrenceToUpdateIndex);
+  }
+  const addedExceptionId = extractExceptionIdFromOccurrenceId(occurrenceId);
+  return exceptions.find(({ key }) => addedExceptionId === key);
+};
+
+export const getExceptionChangeGroups = (
+  pacedTrainExceptions: OccurrenceException
+): ExceptionChangeGroup[] =>
+  Object.keys(pacedTrainExceptions).map(
+    (exception) => exceptionChangeGroupsDict[exception as keyof OccurrenceException]
+  );
+
+export const formatPacedTrainWithOccurenceDetails = (
+  pacedTrain: PacedTrainWithDetails,
+  exceptions: OccurrenceException
+) => {
+  const updatedPacedTrain: PacedTrainWithDetails & { rollingStockName?: string } = pacedTrain;
+  if (exceptions.train_name) {
+    updatedPacedTrain.name = exceptions.train_name.value;
+  }
+  if (exceptions.start_time) {
+    updatedPacedTrain.startTime = new Date(exceptions.start_time.value);
+  }
+  if (exceptions.speed_limit_tag) {
+    // speed limit tag will always be a string or null
+    updatedPacedTrain.speedLimitTag = exceptions.speed_limit_tag.value!;
+  }
+  if (exceptions.labels) {
+    updatedPacedTrain.labels = exceptions.labels.value;
+  }
+  if (exceptions.initial_speed) {
+    updatedPacedTrain.initial_speed = exceptions.initial_speed.value;
+  }
+  if (exceptions.constraint_distribution) {
+    updatedPacedTrain.constraint_distribution = exceptions.constraint_distribution.value;
+  }
+  if (exceptions.rolling_stock_category) {
+    updatedPacedTrain.category = exceptions.rolling_stock_category.value;
+  }
+  if (exceptions.rolling_stock) {
+    if (exceptions.rolling_stock.comfort) {
```

Comment:
Do we need this 2nd if ? I thought that if the silo existed, it must be whole, which means that comfort should always be updated